pkg/api/openapi3.go

```diff
@@ -2,10 +2,20 @@ package api
 
 import (
 	"net/http"
+	"strings"
 
 	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
 )
 
 func openapi3(c *contextmodel.ReqContext) {
-	c.HTML(http.StatusOK, "openapi3", nil)
+	data := map[string]interface{}{
+		"Nonce": c.RequestNonce,
+	}
+
+	// Add CSP for unpkg.com to allow loading of Swagger UI assets
+	existingCSP := c.Resp.Header().Get("Content-Security-Policy")
+	newCSP := strings.Replace(existingCSP, "style-src", "style-src https://unpkg.com/", 1)
+	c.Resp.Header().Set("Content-Security-Policy", newCSP)
```

Comment:
If there is no CSP defined, Header().Get() will return "" and we'll end up setting an empty CSP. I don't think that's a problem, but maybe we can avoid it?

```suggestion
	if existingCSP := c.Resp.Header().Get("Content-Security-Policy"); existingCSP != "" {
		newCSP := strings.Replace(existingCSP, "style-src", "style-src https://unpkg.com/", 1)
		c.Resp.Header().Set("Content-Security-Policy", newCSP)
	}
```